### ISSUE TITLE
Ambiguous Call to AddTypes() 

### ISSUE THREAD

**Issue by danielmackay** [ChilliCream/graphql-platform#5935]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

I am trying to use the `AddTypes()` method to auto register my types, but there are conflicting method signatures.  I have used this function as shown in the v13 blog post.

### Steps to reproduce

1. Install the following packages:

```xml
    <PackageReference Include="HotChocolate.AspNetCore" Version="13.0.5" />
    <PackageReference Include="HotChocolate.Data.EntityFramework" Version="13.0.5" />
```

2. Add the following server configuration:

```cs
builder.Services
    .AddGraphQLServer()
    .AddTypes()
```

3. Confirm the following error is shown:

![image](https://user-images.githubusercontent.com/2636640/223429585-8aa6267d-992a-45a5-afd0-bac94010d8bd.png)

### Version

13.0.5

**Comment by glen-84** [comment]
The name of the method is based on the name of your assembly. You can customize it with `[assembly: Module("MyName")]`, and then use `AddMyNameTypes()`, or just use the default name of your assembly.

**Comment by michaelstaib** [comment]
@glen-84  is correct. In our default template we defined the ModuleInfo to add a module called types thats why in the default template we have the AddTypes method.

I am closing this issue as its not a bug but a config issue.

[assembly:Module("MyTypes")] ... will create AddMyTypes

**Comment by danielmackay** [comment]
I did manage to get this working by adding ModuleInfo.cs and a reference to the package to do the source generation.

Would be great to see these extra steps added to the documentation. 

I would be happy to help with this also if you could provide some direction. 🙂

**Comment by michaelstaib** [comment]
@tobias-tengler can you provide some directions on how @danielmackay can add this to the docs?

**Comment by tobias-tengler** [comment]
Sure :) @danielmackay you can find the documentation in this repository under `website`. I would add a new page to `hotchocolate/v13/defining-a-schema`, something like  `automatic-type-registration.md`. There you can add a Titel and the actual content (you can look at other documents for guidance).
How you can add this new page is detailed here: https://github.com/ChilliCream/graphql-platform/tree/main/website#creating-a-new-documentation-page

To preview your changes run `yarn` in the `website` directory, followed by `yarn start`.

If you need any further help, please let new know :)

**Comment by awebCaseys** [comment]
How was this solved? I cannot figure it out and can't find the documentation that @danielmackay described.

**Comment by HunterAhlquist** [comment]
I'm getting the same error on the GraphQL server template. Strangely it still compiles...

**Comment by mariuszbyahoo** [comment]
@michaelstaib I have to confirm what @HunterAhlquist said. I'm using JetBrains Rider if that matters.
Nonetheless, when you will open [your own tutorial](https://chillicream.com/docs/hotchocolate/v15/get-started-with-graphql-in-net-core) then you will not be able to even compile the template generated by the HotChocolate. I am using the following HotChocolate versions: 
`  <ItemGroup>
    <PackageReference Include="HotChocolate.AspNetCore" Version="15.1.1" />
    <PackageReference Include="HotChocolate.AspNetCore.CommandLine" Version="15.1.1" />
    <PackageReference Include="HotChocolate.Types.Analyzers" Version="15.1.1">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
  </ItemGroup>`

Compiler is not marking any line of code as an issue, but once I will try to build, I am getting the following error message:

`0>Program.cs(3,22): Error CS0121 : The call is ambiguous between the following methods or properties: 'SchemaRequestExecutorBuilderExtensions.AddTypes(IRequestExecutorBuilder, params Type[])' and 'SchemaRequestExecutorBuilderExtensions.AddTypes(IRequestExecutorBuilder, params INamedType[])'
`

Time to reopen the issue chaps.

**Comment by lucsoft** [comment]
Yeah also can't get the example running

**Comment by mariuszbyahoo** [comment]
Guys, we can downgrade to 15.0.3 OR upgrade to prerelease to 15.1.2-p.1 which already fixes the source generator. I've tested both (as @michaelstaib  suggested) - and the prerelease version works just fine

**Comment by dave-chow** [comment]
Thanks @mariuszbyahoo. 
Currently I'm using v13 and upgrading to v15.
Can fix the issue with HotChocolate.Types.Analyzers v15.0.3.
But I try to upgrade all Hotchocolate all related library to v15.1.10. Still having the same issue. Is it really fixed in v15.1.2?

**Comment by mariuszbyahoo** [comment]
@dave-chow sorry for a late response but I stepped away even from the donet world recently, I think I will not be able to assist you further
